openedx/core/djangoapps/user_api/accounts/tests/test_retirement_views.py

```diff
@@ -1007,15 +1007,26 @@ def test_move_from_dead_end(self):
 
     def test_move_backward(self):
         """
-        Confirm that trying to move to an earlier step in the process fails
+        Confirm that trying to move to an earlier step in the process fails if not forced
         """
         retirement = UserRetirementStatus.objects.get(id=self.retirement.id)
-        retirement.current_state = RetirementState.objects.get(state_name='COMPLETE')
+        retirement.current_state = RetirementState.objects.get(state_name='LOCKING_ACCOUNT')
         retirement.save()
 
         data = {'new_state': 'PENDING', 'response': 'this should fail'}
         self.update_and_assert_status(data, status.HTTP_400_BAD_REQUEST)
 
+    def test_force_move_backward(self):
+        """
+        Confirm that trying to move to an earlier step in the process succeeds if forced
+        """
+        retirement = UserRetirementStatus.objects.get(id=self.retirement.id)
+        retirement.current_state = RetirementState.objects.get(state_name='COMPLETE')
```

Comment:
Why is this test starting at the `COMPLETE` state while the one above was changed to use `LOCKING_ACCOUNT`? Shouldn't they both start at the same state? And - if the starting states can be the same, can the tests be collapsed into a single DDT-ized test?

Reply:
There are different code paths for moving backwards vs. moving from a dead end state. This started off as a budget attempt to try to hit both, but I'll go back and DDT all of the permutations.